providers/base/bin/check_prerelease.py

```diff
@@ -27,72 +27,135 @@
 Usage:
    check-prerelease.py
 """
-
+import logging
+import os
 import platform
 import shlex
 import sys
 
 from subprocess import CalledProcessError, check_output
 
 
-def check_kernel_status():
+def get_apt_cache_information(command: str):
+    """ Execute the given apt-cache command and return the information.
+
+    This function runs the specified apt-cache command using the `check_output`
+    function, which returns the information about the Linux kernel package
+    queried by the command.
+
+    :param command: A string representing the apt-cache command to be executed.
+
+    :return:
+        A string containing the information retrieved from the apt-cache
+        command.
+
+    :raises CalledProcessError:
+        If the apt-cache command returns an empty string with exit code 0,
+        indicating a non-existent package.
+    :raises SystemExit:
+        If the apt-cache command returns an error status, indicating that
+        the kernel does not match any installed package.
+    """
+    try:
+        aptinfo = check_output(shlex.split(command), universal_newlines=True)
+        # "apt-cache showpkg" returns an empty string with exit code 0 if
+        # called on a non-existent package.
+        if aptinfo == '':
+            raise CalledProcessError(returncode=None, cmd=None)
+        return aptinfo
+    except CalledProcessError:
+        # "apt-cache show" returns an error status if called on a
+        # non-existent package.
+        logging.error(
+            "* Kernel does not match any installed package!")
```

Comment:
```suggestion
        if aptinfo == "":
            raise CalledProcessError(returncode=1, cmd=command)
        return aptinfo
    except CalledProcessError as e:
        # "apt-cache show" returns an error status if called on a
        # non-existent package.
        logging.error(e)
        logging.error(
            "* Kernel does not match any installed package!")
```

I feel like it's better to raise `CalledProcessError` with a proper returncode and command; and we can actually use this to show more information in the output (with `logging.error(e)`), which could help if anything goes wrong during testing.

Reply:
Nice suggestion, I like it.